sentry/src/main/java/io/sentry/SendCachedEnvelopeFireAndForgetIntegration.java

```diff
@@ -84,6 +91,25 @@ public final void register(final @NotNull IHub hub, final @NotNull SentryOptions
                 }
               });
 
+      final String dirPath = factory.getDirPath();
+      if (dirPath != null && EnvelopeCache.hasStartupCrashMarker(dirPath, options)) {
+        options
+          .getLogger()
+          .log(SentryLevel.DEBUG, "Startup Crash marker exists, blocking flush.");
+        try {
+          future.get(options.getStartupCrashFlushTimeoutMillis(), TimeUnit.MILLISECONDS);
```

Comment:
So this is blocking the execution of this integration, but there's a list of integrations, I am wondering if the order here matters now, let's say this is the 1st integration, all the following integrations will have to wait, which is fine, my question is, does it make sense to install the most important integrations first before blocking this one? (such as Uncaught exception).

Reply:
Yeah we already install them in the SentryOptions ctor before everything else runs, right?  https://github.com/getsentry/sentry-java/blob/54cebc81e18ff2f50da062cac751681e7e4e83d4/sentry/src/main/java/io/sentry/SentryOptions.java#L1822-L1844

Reply:
Sounds good, I'd also make sure that `AnrIntegration` is after the blocking calls, we don't want to trigger ANRs if that's the intention.

Reply:
yep, it's installed afterwards, all good here 👍